muspy/base.py

```diff
@@ -753,11 +735,23 @@ def sort(
         Object itself.
 
         """
-        if attr is None:
-            for attribute in self._list_attributes:
-                self._sort(attribute, recursive)
-        elif attr in self._list_attributes:
-            self._sort(attr, recursive)
-        else:
-            raise TypeError("`{}` must be a list attribute.")
+        attr_type: Type
+        value: list
+        for _, attr_type, value in self._traverse_lists(attr, recursive):
+            if value and "time" in getattr(attr_type, "_attributes"):
+                value.sort(key=attrgetter("time"))
+
         return self
+
+    @property
+    def flat(self) -> Iterable:
+        """A flat representation of this object. Iterating over it
+        yields all items in all list attributes inside this object
+        (recursively). Non-list attributes are not included.
+        """
+        return self._flat
+
+    def _flat_generator(self) -> Iterable:
+        value: list
+        for _, _, value in self._traverse_lists(attr=None, recursive=True):
+            yield from value
```

Comment:
Right, I didn't realize that.

How about this? `flat` is now reusable, and it's always the same object, so it's not a problem to have it as a property.

If you still insist on it being a function, why not call it e.g. `walk_lists`?

Reply:
I just want it to behave more like `dict.items()` or `dict.values()` rather than `numpy.ndarray.flat` or `music21.stream.Stream.flat`. It's quite common in Python to have a function that returns an iterator.

Reply:
True. Is `walk_lists()` OK then? The name should give a hint that the return value is good only for (one) iteration.

BTW `dict.items()` and `dict.values()` also return objects that support more than just iteration.

Reply:
Anyway, let's not overthink this. `lists()` with an optional `recursive` parameter (maybe `False` by default?) is OK too.

Reply:
Actually, `lists()` suggests that it would iterate over lists. But my `flat` iterates over list items, not lists themselves. So we need a better name.

Reply:
I made it `list_items()` and added `list()` to iterate over the actual lists.